sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -207,13 +207,18 @@ export class EducationProgramOfferingControllerService {
    * @param offeringValidations validations to be verified.
    * @param csvModels reference CSV models to provide extra context in case
    * an error must be generated.
+   * @param considerWarningsAsErrors if true, a warning will be considered as an error also,
+   * otherwise only errors in the errors array will be considered errors.
    */
   assertOfferingsValidationsAreValid(
     offeringValidations: OfferingValidationResult[],
     csvModels: OfferingCSVModel[],
+    considerWarningsAsErrors = false,
```

Comment:
In what scenario would we consider the warnings as error?

Reply:
During the _validation-only_ scenario. The idea is to provide errors and warnings to the user during the validation.
One not in _validation-only_ with all "errors" are actually only warnings the process in considered successful and the records re inserted in the "Creation pending" status.
The idea of `considerWarningsAsErrors` is to tell the method that either critical errors or warnings should be treated in the same way throwing an exception if any exists.